### PR TITLE
SNOW-871839: Fix snyk permissions

### PR DESCRIPTION
**Pull request by sfc-gh-pbulawa**
### Description
Fix snyk permissions.

### Checklist
- [x] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [x] Create tests which fail without the change (if possible)
- [x] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [x] Extend the README / documentation and ensure is properly displayed (if necessary)
- [x] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

File: .github/workflows/snyk-pr.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ on:
 jobs:
   whitesource:
     runs-on: ubuntu-latest
+    permissions: write-all
```
Comment (sfc-gh-igarish) [inline .github/workflows/snyk-pr.yml:9]: what's the issue here? normally prod-sec take care of snyk.

Comment (sfc-gh-pbulawa) [reply .github/workflows/snyk-pr.yml:9]: When troubleshooting .NET Snyk PR [#690](https://github.com/snowflakedb/snowflake-connector-net/pull/690) we have found that Snyk did not have the permissions to comment/modify the title of the PR thus the `snyk-pr / whitesource` job constantly failed.

Adding this permissions resolved the issue.

Comment (sfc-gh-dprzybysz) [reply .github/workflows/snyk-pr.yml:14]: do we need proc-sec approval on this PR?

File: .github/workflows/snyk-pr.yml [hunk 2]
```diff
@@ -3,29 +3,36 @@ on:
...
-    - name: PR
-      uses: ./whitesource-actions/snyk-pr
-      env:
-        PR_TITLE: ${{ github.event.pull_request.title }}
-      with:
-        jira_token: ${{ secrets.JIRA_TOKEN_PUBLIC_REPO }}
-        gh_token: ${{ secrets.GITHUB_TOKEN }}
-        amend: false # true if you want the commit to be amended with the JIRA number
+      - name: PR
+        uses: ./whitesource-actions/snyk-pr
```
[the remark this answers is not in the document]
Comment (sfc-gh-pbulawa) [reply .github/workflows/snyk-pr.yml:32]: In my opinion reformatting with a change is not a big deal and it helps the code to be in-spec.

File: .github/workflows/snyk-pr.yml (hunk 1 above)
Comment (sfc-gh-pbulawa) [reply .github/workflows/snyk-pr.yml:14]: This change has been copied from their change in `snowflake-connector-net` driver.

**Review by sfc-gh-dprzybysz — APPROVED** [review]

File: .github/workflows/snyk-pr.yml (hunk 1 above)
Comment (sfc-gh-igarish) [reply .github/workflows/snyk-pr.yml:14]: @sfc-gh-jfan please check.

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

File: .github/workflows/snyk-pr.yml (hunk 1 above)
Comment (sfc-gh-jfan) [reply .github/workflows/snyk-pr.yml:14]: it looks good to me, it should be able to fix the issue here

**Pull request merged**